tests/backup/backup_upgrade_test.go

```diff
@@ -204,3 +206,419 @@ var _ = Describe("{StorkUpgradeWithBackup}", func() {
+				log.InfoD("Creating schedule backup with schedule [%s] of source cluster namespace [%s]", singleNSScheduleName, namespace)
+				err = CreateScheduleBackup(singleNSScheduleName, SourceClusterName, backupLocationName, backupLocationUid, []string{namespace},
+					labelSelectors, orgID, preRuleNames[appName], preRuleUids[appName], postRuleNames[appName], postRuleUids[appName], schedulePolicyName, schedulePolicyUid, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of schedule backup with schedule [%s]", singleNSScheduleName))
+				firstScheduleBackupName, err := GetFirstScheduleBackupName(ctx, singleNSScheduleName, orgID)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching name of the first schedule backup with schedule [%s]", singleNSScheduleName))
+				log.Infof("First schedule backup name: [%s]", firstScheduleBackupName)
+				singleNamespaces = SafeAppend(&mutex, singleNamespaces, namespace).([]string)
+				singleNSScheduleNames = SafeAppend(&mutex, singleNSScheduleNames, singleNSScheduleName).([]string)
+			}
+			_ = TaskHandler(Inst().AppList, createSingleNSBackupTask, Parallel)
+		})
+		Step("Create schedule backup of all namespaces", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			allNamespaces = make([]string, 0)
+			for _, appName := range Inst().AppList {
+				allNamespaces = append(allNamespaces, dstClusterAppNamespaces[appName]...)
+			}
+			allNSScheduleName = fmt.Sprintf("%s-all-ns-schedule-%v", BackupNamePrefix, time.Now().Unix())
+			labelSelectors := make(map[string]string)
+			log.InfoD("Creating schedule backup with schedule [%s] of all namespaces of destination cluster [%s]", allNSScheduleName, allNamespaces)
+			err = CreateScheduleBackup(allNSScheduleName, destinationClusterName, backupLocationName, backupLocationUid, allNamespaces,
+				labelSelectors, orgID, "", "", "", "", schedulePolicyName, schedulePolicyUid, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of schedule backup with schedule [%s]", allNSScheduleName))
+			firstScheduleBackupName, err := GetFirstScheduleBackupName(ctx, allNSScheduleName, orgID)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching name of the first schedule backup with schedule [%s]", allNSScheduleName))
+			log.Infof("First schedule backup name: [%s]", firstScheduleBackupName)
+		})
+		Step("Upgrading px-backup", func() {
+			latestPxBackupVersionFromEnv := os.Getenv("LATEST_PXBACKUP_VERSION")
+			if latestPxBackupVersionFromEnv == "" {
+				latestPxBackupVersionFromEnv = latestPxBackupVersion
+			}
+			log.InfoD("Upgrading px-backup to latest version [%s]", latestPxBackupVersionFromEnv)
+			err := UpgradePxBackup(latestPxBackupVersionFromEnv)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying completion of px-backup upgrade to latest version [%s]", latestPxBackupVersionFromEnv))
+		})
+		Step("Restore full backups created with and without pre- and post-exec rules", func() {
```

Comment:
How are you making sure it's a full backup?

Reply:
Done. Added a helper function IsFullBackup to check if the backup is full backup or not.